src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -151,6 +151,55 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
     
+    // Add all WSL distros from: HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+    Profile WSLDistro{};
+    HKEY hLXSSKey;
+    // Should I use RegOpenCurrentUser to access the uncached profile?
+    // https://docs.microsoft.com/en-us/windows/desktop/api/winreg/nf-winreg-regopencurrentuser#remarks
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\Lxss",
+        0, KEY_READ, &hLXSSKey) == ERROR_SUCCESS)
+    {
+        std::filesystem::path AppsDirectory = ExpandEnvironmentVariableString(L"%ProgramFiles%");
+        AppsDirectory /= "WindowsApps";
+        std::filesystem::path WSLCommandLine = L"";
+
+        wchar_t WSLKey[MAX_PATH] = L""; // Registry key for a distinct WSL Distribution
+        wchar_t DistributionName[MAX_PATH] = L""; // The name to be displayed by the flyout list
+        wchar_t PackageFamilyName[MAX_PATH] = L""; // The distinct name for the distribution
+
+        // These tell the Registry functions how much space they have to work with when returning a value.
+        DWORD WSLKey_Length = MAX_PATH;
+        DWORD DistributionName_Length = MAX_PATH;
+        DWORD PackageFamilyName_Length = MAX_PATH;
+
+        for (int i = 0;
+            RegEnumKeyExW(hLXSSKey, i, WSLKey, &WSLKey_Length, NULL, NULL, NULL, NULL) == ERROR_SUCCESS;

```

Comment:
Use nullptr instead where possible throughout this file, per this ISO C++ Guideline:
[ES.47: Use nullptr rather than 0 or NULL](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es47-use-nullptr-rather-than-0-or-null)

Reply:
Updated.